Session.cc

```diff
@@ -961,6 +962,13 @@ void Session::OnCatalogFilter(CARTA::CatalogFilterRequest filter_request, uint32
     });
 }
 
+void Session::OnSpectralLineRequest(CARTA::SpectralLineRequest spectral_line_request, uint32_t request_id) {
+    CARTA::SpectralLineResponse spectral_line_response;
+    carta::SpectralLineRequest requester;
```

Comment:
I think this is very confusing. `CARTA::SpectralLineRequest` and `carta::SpectralLineRequest` are very different objects. I'm not even sure why you need to create an instance of this, rather than just using the static functions. The `carta::SpectralLineRequest` class doesn't have any instance variables. Maybe `carta::SpectralDbQuery` or something like that would make more sense .